Oracle.md

```diff
@@ -0,0 +1,197 @@
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of adaptors signatures for some CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Three kinds of event outcomes are defined (note that a single event can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: °C
+precision: 0
+```
+
+### Digit decomposition
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using digit decomposition.
+
+The event descriptor should include:
+
+- base: a number representing the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed (e.g. 2 for binary, 10 for decimal, 16 for hex-decimal).
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base exponent by which to multiply the number represented by the composition of the digits to obtain the actual outcome value.
+- number of digits: the number of digits that the oracle will sign (if `is-signed` is set to true, the number of R-values for the event will be `number of digits + 1`).
+
+In the case where the number of digits that make up the outcome value exceeds the number of r-values that the oracle committed to, the oracle should sign the maximal possible value that it can attest to.
+In the case where the outcome value became negative but the oracle did not provide an extra R-value, it should sign the value 0.
+The minimum and maximum values that can be attested to by an oracle should thus be interpreted as `max_value or more` and `min_value or less`.
+This enables contracting party to specify payouts for the overflow and underflow cases, and avoid having to use the refund path which in most cases would be unfair.
+More complex constructions where considered to handle these, but the simplicity of the currently specified approach is the reason that it was chosen.
```

Comment:
IMO, you could insert a footnote, event to a GH discussion to enlighten future document readers. Otherwise, this kind of sentence is a bit useless as you don't know what "complex constructions where considered".